programs/stake/src/stake_state.rs

```diff
@@ -1152,6 +1172,87 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+enum MergeKind {
+    Inactive(Meta, u64),
+    ActivationEpoch(Meta, Stake),
+    FullyActive(Meta, Stake),
+}
+
+impl MergeKind {
```

Comment:
I think grouping code together like this is a good direction; got a lot easier to read.